smack-experimental/src/main/java/org/jivesoftware/smackx/spoiler/SpoilerManager.java

```diff
@@ -0,0 +1,88 @@
+/**
+ *
+ * Copyright 2018 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.spoiler;
+
+import java.util.Map;
+import java.util.WeakHashMap;
+
+import org.jivesoftware.smack.Manager;
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smackx.disco.ServiceDiscoveryManager;
+import org.jivesoftware.smackx.spoiler.element.SpoilerElement;
+
+public final class SpoilerManager extends Manager {
```

Comment:
I think the manager should probably get auto initialized (see other managers for the corresponding Smack pattern). I am also wondering if it should provide methods to start and stop announcing support for spoiler messages (e.g. `startAnnouncingSupportForSpoilerMessages() and `stopAnnouncingSupportForMessages()`). But then again, I can't come up with a good use case for those methods, given that the manager would, once instantiated, start announcing support.